OneWelcomeExampleApp/Presenters/QRCodePresenter.swift

```diff
@@ -0,0 +1,114 @@
+//  Copyright © 2023 OneWelcome. All rights reserved.
+
+import AVFoundation
+import UIKit
+
+protocol QRCodePresenterProtocol: AnyObject {
+    func setupErrorLabel(text: String)
+    func present(with navigationController: UINavigationController, delegate: QRCodeViewDelegate)
+    func setupCaptureSession(in view: UIView)
+    func startSession()
+    func stopSession()
+    func cancel()
+}
+
+class QRCodePresenter: NSObject, QRCodePresenterProtocol {
+    private let qrCodeViewController: QRCodeViewController
+    private let navigationController: UINavigationController
+    private var captureSession: AVCaptureSession?
+    private var previewLayer: AVCaptureVideoPreviewLayer!
+
+    init(qrCodeViewController: QRCodeViewController, navigationController: UINavigationController) {
+        self.navigationController = navigationController
+        self.qrCodeViewController = qrCodeViewController
+    }
+    
+    func present(with navigationController: UINavigationController, delegate: QRCodeViewDelegate) {
```

Comment:
How about
`func present(controller: UINavigationController, delegate: QRCodeViewDelegate) {` ?
we would avoid variables overlapping and it would be obvious who is presenter and who is going to be presented.